druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -3317,6 +3317,46 @@ private void testCountWithApproxDistinct(boolean approx, String sql, String expe
         .returnsUnordered("C=86829");
   }
 
+  @Test
+  public void testFilterwithFloorOnTime() {
+    // Test filter on floor on time column is pushed to druid
+    final String sql =
+        "Select floor(\"__time\" to MONTH) as t from \"wikiticker\" where "
+            + "floor(\"__time\" to MONTH) between '2014-08-01 00:00:00 UTC'"
+            + "and '2016-10-01 00:00:00 UTC' order by t limit 2";
+
+    final String druidQueryPart1 = "\"filter\":{\"type\":\"and\",\"fields\":"
+        + "[{\"type\":\"bound\",\"dimension\":\"__time\",\"lower\":\"2014-08-01T00:00:00.000Z\","
+        + "\"lowerStrict\":false,\"ordering\":\"lexicographic\","
+        + "\"extractionFn\":{\"type\":\"timeFormat\",\"format\":\"yyyy-MM-dd";
+    final String druidQueryPart2 = "HH:mm:ss.SSS";
+    final String druidQueryPart3 = ",\"granularity\":\"month\",\"timeZone\":\"UTC\","
+        + "\"locale\":\"en-US\"}},{\"type\":\"bound\",\"dimension\":\"__time\""
+        + ",\"upper\":\"2016-10-01T00:00:00.000Z\",\"upperStrict\":false,"
+        + "\"ordering\":\"lexicographic\",\"extractionFn\":{\"type\":\"timeFormat\"";
+    final String druidQueryPart4 = "\"dimensions\":[],\"metrics\":[],\"granularity\":\"all\"";
+
+    sql(sql, WIKI_AUTO2)
+        .queryContains(
+            druidChecker(druidQueryPart1, druidQueryPart2, druidQueryPart3, druidQueryPart4))
+        .returnsOrdered("T=1441065600000", "T=1441065600000");
+  }
+
+  @Test
+  public void testSelectFloorOnTimeWithFilterOnFloorOnTime() {
+    final String sql = "Select floor(\"__time\"  to YEAR) as t from \"wikiticker\" "
+        + "where floor(\"__time\" to YEAR) >= '2006-10-01 00:00:00 UTC' order by t limit 1";
+    final String druidQueryPart1 = "filter\":{\"type\":\"bound\",\"dimension\":\"__time\","
+        + "\"lower\":\"2006-10-01T00:00:00.000Z\",\"lowerStrict\":false,"
+        + "\"ordering\":\"lexicographic\",\"extractionFn\":{\"type\":\"timeFormat\","
+        + "\"format\":\"yyyy-MM-dd";
+    final String druidQueryPart2 = "\"granularity\":\"year\",\"timeZone\":\"UTC\","
+        + "\"locale\":\"en-US\"}},\"dimensions\":[],\"metrics\":[],\"granularity\":\"all\"";
+
+    sql(sql, WIKI_AUTO2).queryContains(druidChecker(druidQueryPart1, druidQueryPart2))
+        .returnsOrdered("T=1420070400000");
```

Comment:
not sure if this makes sense, the query is ` Select floor(\"__time\"  to YEAR)` and the result is `T=1420070400000` isn't we should expect something like `2006-10-01T00:00:00.000Z`? 
Do we have to add a cast? is that the SQL standard or this how Calcite handle such queries? @jcamachor and @nishantmonu51

Reply:
Thanks for pointing this out @b-slim . I think this is because of the lack of interpretation in Avatica for the 'timestamp with local time zone' type. @nishantmonu51 , till it is supported there, could we add a cast to timestamp for the select clause as in other tests?

Reply:
added a cast to timestamp, should I  also open a JIRA for follow up to add support for  'timestamp with local time zone' ?